.github/workflows/release.yaml

```diff
@@ -0,0 +1,56 @@
+name: release
+
+on:
+  workflow_dispatch:
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+          fetch-depth: 0
+
+      - id: extract_release
+        run: |
+          branch_name="${GITHUB_REF#refs/heads/}"
+          if [[ $branch_name =~ ^release-([0-9]+)$ ]]; then
+            release_number=${BASH_REMATCH[1]}
+            echo "release_number=$release_number" >> $GITHUB_ENV
+          else
+            echo "Invalid branch name format"
+            exit 1
+          fi
+
+      - id: git_config
+        run: |
+          git config --global user.name "GitHub Actions"
+          git config --global user.email "actions@github.com"
+
+      - id: push_prod_tag
+        run: |
+          tag_name="${release_number}.1"
+          git tag -a $tag_name -m stack-${release_number}-prod
```

Comment:
I feel like I'm just missing some context:
* How do we build + tag patch versions today? Will this be compatible with that process (or is that an intended future change)?
* What's the purpose of these new `-prod`/`-staging` tags?

Reply:
Now I manually run a shell script that takes the repo, the current and next release number (e.g. 'create-branch ~/bin/create-branch.sh SynapseWebClient 512 513') to tag and create the branches each week. It clones the repo and does essentially the same steps as here (simpler since I give it the versions...). I figured it'd be nice to just click a button right in GH since I'm there re-checking that everything is merged anyway.

The '-prod' and 'staging' are just added to the comment on the git tag (to remind that .0 lives while that version is on staging and .1  while it's in prod).

To version the artifacts, we use 2 scripts (one for dev branch, one for release branches) that modify the version in all the pom.xml files, I would like to replace them by the optional 'mvn versions:set' step in build/actions.yml.